app/code/Magento/CustomerImportExport/Model/Import/Customer.php

```diff
@@ -569,11 +569,11 @@ public function getEntityTable()
      */
     public function getValidColumnNames()
     {
-        $this->validColumnNames = array_merge(
-            $this->validColumnNames,
-            $this->customerFields
+        return array_unique(
+            array_merge(
+                $this->validColumnNames,
+                $this->customerFields
```

Comment:
Such implementation is better than before but still has linear computational complexity.

If you flip the arrays fields will be unique by design and you will be able to perform main operations with `O(1)` complexity.

So, use `array_keys($array)` to get values and `$array1 + $array2` to combine two arrays.